### ISSUE TITLE
Windows installation problem: can't find solver

### ISSUE THREAD

**Issue by pgkirsch** [hoburg/gpkit#758]
(I know there have been a number of Windows installation issues opened previously, but I wasn't able to find a solution to my problem on any of them.)

Trying to install GPkit on a windows machine using Python(x,y), cvxopt, and pip. As @mayork noted in comments on commit f926d46, there seems to be a problem with installing gpkit on Windows using pip. I installed python(x,y), including the cvxopt option, and then pip installed gpkit. I can import both gpkit and cvxopt separately without issue, but gpkit cannot find cvxopt so unit tests fail. Also, there doesn't appear to be a setup.py in my C:\Python27\Lib\site-packages\gpkit. I spoke with @mayork and he recalls having a similar issue with a non-github-cloned copy of gpkit.

**Comment by pgkirsch** [comment]
For the record (and as some already know) installing gpkit by cloning the git repo and running `python setup.py install` works fine.

**Comment by bqpd** [comment]
Hmm, I wonder if we should quickly patch the install instructions to mention that! Following the `debugging installation` section didn't help, I take it?

**Comment by pgkirsch** [comment]
Unfortunately not

**Comment by whoburg** [comment]
+1 for patching the install instructions -- @pgkirsch's comments above match my personal experience with trying to help windows users get up and running.

**Comment by mayork** [comment]
what do these instructions for windows mosek install mean exactly?

```
Make sure gcc is on your system path.
To do this, type gcc into a command prompt.
If you get executable not found, then install the 64-bit version of mingw.
Make sure the mingw bin directory is on your system path (you may have to add it manually).
```
there's a number of different bin directories inside my mingw folder. It seems that no matter what value I set the mingw path to, gcc still returns as an unknown command

**Comment by whoburg** [comment]
@galbramc, do the Jenkins windows machines use mingw? Any ideas on getting `gcc` working on Windows?

**Comment by galbramc** [comment]
If you are using Python(x,y) with cvxopt then you do not need gcc. There is something else that must be wrong. What is in your settings file in C:\Users\your_user_name\.gpkit.

**Comment by mayork** [comment]
I'm trying to solve some models cvxopt can't handle so I tried again to install mosek, and I think there's some gcc problems occurring

**Comment by galbramc** [comment]
Ahh, so you are most likely encountering the problem with the cached settings file as I outline here #679. 

Did you install the 64bit gcc? Is it in your system path? Do you know how to check if it's in the system path?

**Comment by mayork** [comment]
The download at the link in the install instructions only gave me one option for gcc, but it lives by default in program files x86 so I'm confident its 32 bit. I added manually add the mingw bin folder to the system path.

**Comment by galbramc** [comment]
?? You need to install the 64 bit version of mingw. You also need to install a 64 bit version of Python, and we recommend anaconda for this. The 32bit version of mosek does not work on windows for some reason, at least I have not been able to get it to work.

**Comment by mayork** [comment]
Hmm interesting. I believe that at least mosek_cli is working for me right now

**Comment by mayork** [comment]
![image](https://cloud.githubusercontent.com/assets/6796415/16904991/6cd2057e-4c6d-11e6-91f4-b14cac762a27.png)
![image](https://cloud.githubusercontent.com/assets/6796415/16904995/79059914-4c6d-11e6-87a3-0011f4958945.png)

here's some print outs from when I run the tests...it seems weird that the mosek_cli tests are all passing

**Comment by galbramc** [comment]
32 bit? Yeah. The cli might work... because gpkit is using files to communicate with it.

**Comment by mayork** [comment]
ah okay I see

**Comment by galbramc** [comment]
What is in your settings file? Is this 32 bit mosek and python(x,y)?

**Comment by mayork** [comment]
yeah it is

**Comment by galbramc** [comment]
So what is in your settings file in: C:\Users\your_user_name\.gpkit\settings?

**Comment by mayork** [comment]
C:\Users\mayork\.gpkit

I just navigated to that folder and it was empty...

**Comment by galbramc** [comment]
Oh that's right. Its not there. It's somewhere in C:\Python27\Lib\site-packages\gpkit

**Comment by mayork** [comment]
installed_solvers : cvxopt, mosek_cli
mosek_bin_dir : C:\Program Files (x86)\Mosek\7\tools\platform\win32x86\bin

**Comment by galbramc** [comment]
Ok, well it looks like you might be able to use the cli version of mosek then. I don't know if it has any limitations. It is likely slower since it communicates with mosek using files, but I don't know if that matters to you. You have to check with @bqpd and @whoburg if this is enough for you to get your work done.

**Comment by mayork** [comment]
what's weird is it solves some problems but others times throws this error before solving
```
Traceback (most recent call last):
  File "C:\Users\mayork\Documents\GitHub\gpkit-models\gpkitmodels\Commercial Sizing\Commerical_Flight_Profile.py", line 583, in <module>
    sol = m.localsolve(solver="mosek_cli", verbosity = 4, iteration_limit=1000)
  File "c:\users\mayork\documents\github\gpkit\gpkit\constraints\prog_factories.py", line 80, in solvefn
    result = solvefn(solver, verbosity-1, *args, **kwargs)
  File "c:\users\mayork\documents\github\gpkit\gpkit\constraints\signomial_program.py", line 108, in localsolve
    result = gp.solve(solver, verbosity-1, **kwargs)
  File "c:\users\mayork\documents\github\gpkit\gpkit\geometric_program.py", line 182, in solve
    k=self.k, *args, **kwargs)
  File "c:\users\mayork\documents\github\gpkit\gpkit\_mosek\cli_expopt.py", line 103, in imize
    for logline in check_output(["mskexpopt", filename]).split(b"\n"):
  File "C:\Python27\lib\subprocess.py", line 573, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
CalledProcessError: Command '['mskexpopt', 'c:\\users\\mayork\\appdata\\local\\temp\\tmp5corbk\\gpkit_mosek']' returned non-zero exit status 1432
```

**Comment by galbramc** [comment]
This is definitely a question for @bqpd and @whoburg. Thought that non-zero exit status likely means that mosek simply failed for some reason. Did mosek print anything to the screen?

Can you create a unit test that reproduces the problem to help them hunt down the bug?

**Comment by mayork** [comment]
nothing was printed to the screen outside of that. I'll push some commits tomorrow demonstrating working and non-working cases

**Comment by galbramc** [comment]
If you can actually make a unit test then they can make sure the error never occurs again (assuming it's a fixable bug).

**Comment by mayork** [comment]
I got the gcc thing sorted out, although I still have the expopt problem which is a previously known problem. I'm not sure why some models will solve with mosek_cli and others won't, so I don't know how to test it. I'll investigate this more in a few weeks.

**Comment by galbramc** [comment]
What did you do to sort out gcc? You get an export.dll shared library now, but calling it from gpkit fails? That was what I had seen in the past.

**Comment by mayork** [comment]
i just retired adding each of the two bin god mets to the system path. Not sure what went wrong the first time.

**Comment by galbramc** [comment]
Ok, but you get an error when gpkit tries to use expopt?

**Comment by mayork** [comment]
Yeah exactly

**Comment by galbramc** [comment]
Ok. It might be possible to fix this bug, but that is up too @bpqd and @whoburg. The only solution I know is to use only 64bit, but you loose cvxopt of you do that.

**Comment by bqpd** [comment]
I think cvxopt works with 64 bit anaconda now, right?

**Comment by galbramc** [comment]
That's news to me if it does. We don't have it on any testing machine.

**Comment by bqpd** [comment]
see https://github.com/hoburg/gpkit/issues/811

**Comment by galbramc** [comment]
Ok. Let me give it a try on the vm. I won't be able to until Monday though.

**Comment by bqpd** [comment]
resolved by recent documentation, as #934
